packages/block-editor/src/components/block-tools/block-selection-button.js

```diff
@@ -69,6 +71,26 @@ function BlockSelectionButton( { clientId, rootClientId } ) {
 			const orientation =
 				getBlockListSettings( rootClientId )?.orientation;
 			const match = getActiveBlockVariation( name, attributes );
+			const isBlockTemplatePart =
+				blockType?.name === 'core/template-part';
+
+			let isNextBlockTemplatePart = false;
+			const nextClientId = getNextBlockClientId();
+			if ( nextClientId ) {
+				const { name: nextName } = getBlock( nextClientId );
+				const nextBlockType = getBlockType( nextName );
+				isNextBlockTemplatePart =
+					nextBlockType?.name === 'core/template-part';
```

Comment:
I think we should check if it's a header / footer template area too. This would stop us moving all blocks either side of any template part.

Reply:
The area check was my initial idea but it's so weirdly saved as a piece of info. Then I realised all template parts should be stuck in place in composer mode.